toolkit/offchain/src/csl.rs

```diff
@@ -35,10 +43,80 @@ fn language_kind_to_u8(language: LanguageKind) -> u8 {
 	}
 }
 
+/// Creates a CSL [`TransactionBuilderConfig`] for given [`ProtocolParametersResponse`].
+/// This function is not unit-testable because [`TransactionBuilderConfig`] has no public getters.
+pub(crate) fn get_builder_config(
+	protocol_parameters: &ProtocolParametersResponse,
+) -> Result<TransactionBuilderConfig, JsError> {
+	TransactionBuilderConfigBuilder::new()
+		.fee_algo(&linear_fee(protocol_parameters))
+		.pool_deposit(&convert_value(&protocol_parameters.stake_pool_deposit)?.coin())
+		.key_deposit(&convert_value(&protocol_parameters.stake_credential_deposit)?.coin())
+		.max_value_size(protocol_parameters.max_value_size.bytes)
+		.max_tx_size(protocol_parameters.max_transaction_size.bytes)
+		.ex_unit_prices(&ExUnitPrices::new(
+			&ratio_to_unit_interval(&protocol_parameters.script_execution_prices.memory),
+			&ratio_to_unit_interval(&protocol_parameters.script_execution_prices.cpu),
+		))
+		.coins_per_utxo_byte(&protocol_parameters.min_utxo_deposit_coefficient.into())
+		.build()
+}
+
+fn linear_fee(protocol_parameters: &ProtocolParametersResponse) -> LinearFee {
+	let constant: BigNum = protocol_parameters.min_fee_constant.lovelace.into();
+	LinearFee::new(&protocol_parameters.min_fee_coefficient.into(), &constant)
+}
+
+fn ratio_to_unit_interval(ratio: &fraction::Ratio<u64>) -> UnitInterval {
+	UnitInterval::new(&(*ratio.numer()).into(), &(*ratio.denom()).into())
+}
+
+/// Coverts ogmios value to CSL value.
+/// It could fail if the input contains negative values, for example ogmios values representing burn.
+pub(crate) fn convert_value(value: &OgmiosValue) -> Result<Value, JsError> {
+	if value.native_tokens.len() > 0 {
+		let mut multiasset = MultiAsset::new();
+		value.native_tokens.iter().try_for_each(|(policy_id, assets)| {
```

Comment:
I think we can do the same thing using a simple `for`. A little more readable imo and avoids returning explicit `Ok(())`

Reply:
You are right. Fixed.